### PR TITLE
Add new perf job to Jenkins

### PR DESCRIPTION
**Pull request by DrewScoggins**
I have added a new perf.groovy file to deal with the perf runs for
CoreFX.  Currently we are only adding Windows, but will add Linux
support after this is running.

### CODE REVIEW

**Comment by DrewScoggins** [comment]
test ci please

**Comment by DrewScoggins** [comment]
test ci please

**Comment by DrewScoggins** [comment]
PTAL @jorive @brianrob

**Comment by DrewScoggins** [comment]
test ci please

File: perf.groovy [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    {
+                        stringParam('BenchviewCommitName', '\${ghprbPullTitle}', 'The name that you will be used to build the full title of a run in Benchview.  The final name will be of the form <branch> private BenchviewCommitName')
+                    }
+                }
+                def configuration = 'Release'
+                def runType = isPR ? 'private' : 'rolling'
+                def benchViewName = isPR ? 'corefx private %BenchviewCommitName%' : 'corefx rolling %GIT_BRANCH_WITHOUT_ORIGIN% %GIT_COMMIT%'
+
+                steps {
+                    //We need to specify the max cpu count to be one as we do not want to be executing performance tests in parallel
```
Comment (brianrob) [inline perf.groovy]: Not sure where you specify CPU count, but the harness won't allow you to run in parallel.  It serializes all execution.

File: perf.groovy [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            // Add the unit test results
+            Utilities.addXUnitDotNETResults(newJob, 'bin/tests/**/Perf-*.xml')
+            def archiveContents = "msbuild.log"
+            
+            // Add archival for the built data.
+            Utilities.addArchival(newJob, archiveContents)
+            // Set up triggers
+            if (isPR) {
+                TriggerBuilder builder = TriggerBuilder.triggerOnPullRequest()
+                builder.setGithubContext("${os} Perf Tests")
```
Comment (brianrob) [inline perf.groovy]: Nit: Change Perf to Performance.

File: perf.groovy [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+def project = GithubProject
+// The input branch name (e.g. master)
+def branch = GithubBranchName
+// Folder that the project jobs reside in (project/branch)
+def projectFolder = Utilities.getFolderName(project) + '/' + Utilities.getFolderName(branch)
+
+// Globals
+
+// Map of os -> osGroup.
+def osGroupMap = ['Ubuntu14.04':'Linux',
```
Comment (brianrob) [inline perf.groovy]: Are all these dictionaries for use in a subsequent PR?  I'm assuming that they are since this PR only enables Windows.

Comment (DrewScoggins) [inline perf.groovy]: Yeah, this was just boiler plate I got from the netci.groovy that we will need when we eventually add other OSes.

File: perf.groovy (hunk 1 above)
Comment (DrewScoggins) [inline perf.groovy]: This is to stop the build from running in parallel.  I set it below with /m:1.  I will move this comment down to where we actually make the call.

File: perf.groovy (hunk 2 above)
Comment (DrewScoggins) [inline perf.groovy]: OK

File: perf.groovy (hunk 1 above)
Comment (brianrob) [inline perf.groovy]: Gotcha.  This is at a higher-level than the runner.  Thanks.

File: perf.groovy (hunk 3 above)
Comment (brianrob) [inline perf.groovy]: Sounds good

File: perf.groovy (hunk 2 above)
Comment (brianrob) [inline perf.groovy]: Thanks

**Comment by DrewScoggins** [comment]
PTAL @brianrob @dpodder @jorive

**Comment by brianrob** [comment]
@DrewScoggins, latest changes LGTM.

**Comment by dpodder** [comment]
CC @jorive

**Pull request merged**